skopt/optimizer/optimizer.py

```diff
@@ -176,7 +183,109 @@ def _check_arguments(self, base_estimator, n_random_starts, acq_optimizer):
+            optimizer.gains_ = np.copy(self.gains_)
+
+        if self.Xi:
+            optimizer.tell(self.Xi, self.yi)
+
+        return optimizer
+
+    def ask(self, n_points=None, strategy="cl_min"):
+        """Query point or multiple points at which objective should be evaluated.
+
+        * `n_points` [int or None, default=None]:
+            Number of points returned by the ask method.
+            If the value is None, a single point to evaluate is returned.
+            Otherwise a list of points to evaluate is returned of size
+            n_points. This is useful if you can evaluate your objective in
+            parallel, and thus obtain more objective function evaluations per
+            unit of time.
+
+        * `strategy` [string, default=`"cl_min"`]:
+            Method to use to sample multiple points (see also `n_points`
+            description). This parameter is ignored if n_points = None.
+            Supported options are `"cl_min"`, `"cl_mean"` or `"cl_max"`.
+
+            - If set to `"cl_min"`, then constant liar strtategy is used
+               with lie objective value being minimum of observed objective
+               values. `"cl_mean"` and `"cl_max"` means mean and max of values
+               respectively. For details on this strategy see:
+
+               https://hal.archives-ouvertes.fr/hal-00732512/document
+
+               With this strategy a copy of optimizer is created, which is
+               then asked for a point, and the point is told to the copy of
+               optimizer with some fake objective (lie), the next point is
+               asked from copy, it is also told to the copy with fake
+               objective and so on. The type of lie defines different
+               flavours of `cl_x` strategies.
+
+        """
+        if n_points is None:
+            return self._ask()
+
+        supported_strategies = ["cl_min", "cl_mean", "cl_max"]
+
+        if not (isinstance(n_points, int) and n_points > 0):
+            raise ValueError(
+                "n_points should be int > 0, got " + str(n_points)
+            )
+
+        if strategy not in supported_strategies:
+            raise ValueError(
+                "Expected parallel_strategy to be one of " +
+                str(supported_strategies) + ", " + "got %s" % strategy
+            )
+
+        # Caching the result. If some new parameters are provided to the ask,
+        # the cache is not used.
+        if (n_points, strategy) in self.cache:
+            return self.cache[(n_points, strategy)]
+
+        # Copy of the optimizer is made in order to manage the
+        # deletion of points with "lie" objective (the copy of
+        # oiptimizer is simply discarded)
+        opt = self.copy()
+
+        X = []
+        for i in range(n_points):
+            x = opt.ask()
+            X.append(x)
+            if strategy == "cl_min":
+                y_lie = np.min(opt.yi) if opt.yi else 0.0  # CL-min lie
```

Comment:
Hmm it might happen that the `n_points` could be something large (eg 64 if you run computations on a 64 core machine), so I  think situation when `n_random_starts` < `n_points` should not raise the error. 
Normally `yi` is empty when no points were generated yet. Do you suggest to make a distinction between the case when `n_random_starts` has been exhaused or not? Right now I do not make this distinction explicitly, but allow this to happen in `_ask` method, hence I am also checking if the list `yi` is empty or not.